api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,344 @@
+	//
+	PublicSubnet *AWSResourceReference `json:"publicSubnet,omitempty"`
+
+	// Zone is the availability zone to be used, a subnet will be created if one is not provided.
+	// The availability zones must be a memeber of the spec.platform.aws.region
+	//
+	Zone string `json:"zone"`
+}
+
+// HostedClusterInfrastructure defines the observed state of HostedClusterInfrastructure
+type HostedClusterInfrastructureStatus struct {
+	// Track the conditions for each step in the desired curation that is being
+	// executed as a job
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// Infrastructure is populated with the created or discovered infrastructure resouce IDs that are
+	// used when creating HostedCluster and  NodePool
+	Infrastructure HostedClusterInfrastructureConfigStatus `json:"infrastructure,omitempty"`
+}
+
+// AWSPlatformInfraIndirectCosnfigStatus specifies configuration for clusters running on Amazon Web Services.
+type AWSPlatformInfraIndirectConfigStatus struct {
+
+	// DhcpOptionsID is the AWS DHCP Options resource ID, must exist and be associated with the VPC
+	// +optional
+	DhcpOptionsID string `json:"dhcpOptionsID"`
+
+	// InternetGatewayID tracks the internet gateway used to communitate with the Internet,
+	// must be attached to the VPC
+	// +optional
+	InternetGatewayID string `json:"internetGatewayID"`
+
+	// NatGatewayID tracks the NAT gateway used for private internal communication
+	// +optional
+	NatGatewayID string `json:"natGatewayID"`
+
+	// PublicEndpointRouteTableID points to the routing table for public routing
+	// +optional
+	PublicEndpointRouteTableID string `json:"publicEndpointRouteTableIds"`
+
+	// PrivateEndpointRouteTableIds points to the routing tables for private routing
+	// +optional
+	PrivateEndpointRouteTableIds []*string `json:"privateEndpointRouteTableIds"`
+
+	// VPCEndpointID points to an endpoint used by S3
+	// +optional
+	VPCEndpointID string `json:"vpcEndpointId"`
+
+	// ElasticIpID points to an Elastic IP used by the Internet Gateway
+	// +optional
+	ElasticIpID string `json:"elasticIpID"`
+
+	// LocalZoneID is the Hosted Zone ID where all the DNS records that are
+	// locally accessible to the cluster exist.
+	//
+	// +optional
+	// +immutable
+	LocalZoneID string `json:"localZoneID,omitempty"`
+
+	// OIDCProviderARN is an ARN value referencing a role appropriate for the Provider.
+	//
+	// The following is an example of a valid policy document:
+	//
+	// {
+	//	"Version": "2012-10-17",
+	//	"Statement": [
+	//		{
+	//			"Effect": "Allow",
+	//			"Action": [
+	//				"iam:CreateARN"
+	//			],
+	//			"Resource": [
+	//				"arn:aws:iam:::oidc-provider",
+	//			]
+	//		}
+	//	]
+	// }
+	OIDCProviderARN string `json:"oidcProviderARN"`
+}
+
+// +genclient
+
+// HostedClusterInfrastructure is the primary representation of a HyperShift cluster's infrastructure.
+// It encapsulates resource that can be created orthogonal to the control plane. Creating a HostedClusterInfrastructure
```

Comment:
>// It encapsulates resource that can be created orthogonal to the control plane. Creating a HostedClusterInfrastructure

typo "resource"

Should we just say: "It creates the infrastructure required for a guest cluster"

Reply:
Used "It creates the infrastructure required for a hosted cluster"